tls/s2n_handshake.c

```diff
@@ -100,3 +101,63 @@ int s2n_handshake_get_hash_state(struct s2n_connection *conn, s2n_hash_algorithm
     }
     return 0;
 }
+
+int s2n_handshake_require_all_hashes(struct s2n_handshake *handshake)
+{
+    memset(handshake->required_hash_algs, 1, sizeof(handshake->required_hash_algs));
+    return 0;
+}
+
+static int s2n_handshake_require_hash(struct s2n_handshake *handshake, s2n_hash_algorithm hash_alg)
+{
+    handshake->required_hash_algs[hash_alg] = 1;
+    return 0;
+}
+
+uint8_t s2n_handshake_is_hash_required(struct s2n_handshake *handshake, s2n_hash_algorithm hash_alg)
+{
+    return handshake->required_hash_algs[hash_alg];
+}
+
+/* Update the required handshake hash algs depending on current handshake session state.
+ * This function must called at the end of a handshake message handler. Additionally it must be called after the
+ * ClientHello or ServerHello is processed in client and server mode respectively. The relevant handshake parameters
+ * are not available until those messages are processed.[M#processedF
```

Comment:
It looks like one little typo made it in - remove `[M#processedF`.

Reply:
sloppy sloppy. Good catch!

Reply:
Fixed in latest update